src/pages/settings/Security/AddDelegate/DelegateMagicCodePage.tsx

```diff
@@ -0,0 +1,84 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useEffect, useRef} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import ScreenWrapper from '@components/ScreenWrapper';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import type {SettingsNavigatorParamList} from '@libs/Navigation/types';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import type CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import ValidateCodeForm from './ValidateCodeForm';
+import type {ValidateCodeFormHandle} from './ValidateCodeForm/BaseValidateCodeForm';
+
+type DelegateMagicCodePageProps = StackScreenProps<SettingsNavigatorParamList, typeof SCREENS.SETTINGS.DELEGATE.DELEGATE_CONFIRM>;
+
+function DelegateMagicCodePage({route}: DelegateMagicCodePageProps) {
+    const {translate} = useLocalize();
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const accountID = Number(route.params.accountID);
+    const role = route.params.role as ValueOf<typeof CONST.DELEGATE_ROLE>;
+
+    const delegatePersonalDetails = PersonalDetailsUtils.getPersonalDetailsByIDs([accountID], -1)[0];
+
+    const optimisticDelegateEmail = account?.delegatedAccess?.delegates?.find((delegate) => delegate.optimisticAccountID === accountID)?.email;
+    const optimisticDelegateData = OptionsListUtils.getUserToInviteOption({
+        searchValue: optimisticDelegateEmail ?? '',
+    });
+
+    const login = delegatePersonalDetails?.login ?? optimisticDelegateData?.login ?? '';
+
+    const styles = useThemeStyles();
+    const validateCodeFormRef = useRef<ValidateCodeFormHandle>(null);
+
+    const currentDelegate = account?.delegatedAccess?.delegates?.find((d) => d.email === login);
+
+    useEffect(() => {
+        if (!currentDelegate || !!currentDelegate.pendingFields?.email || !!currentDelegate.pendingFields?.role || !!currentDelegate.errorFields?.addDelegate) {
```

Comment:
We're so closeeeeee - I'm getting another issue with new accounts, those that aren't contacts already. The modal does not close after adding a fresh contact as a delegate. It works fine for existing contacts.

Reply:
interesting i had fixed it but let me try again